packages/peregrine/lib/store/actions/checkout/asyncActions.js

```diff
@@ -112,36 +94,38 @@ export const getShippingMethods = () => {
 };
 
 export const submitPaymentMethodAndBillingAddress = payload =>
-    async function thunk(dispatch, getState) {
-        submitBillingAddress(payload.formValues.billingAddress)(
-            dispatch,
-            getState
-        );
-        submitPaymentMethod(payload.formValues.paymentMethod)(
-            dispatch,
-            getState
-        );
+    async function thunk(dispatch) {
+        const { countries, formValues } = payload;
+        const { billingAddress, paymentMethod } = formValues;
+
+        dispatch(submitBillingAddress({ billingAddress, countries }));
+        dispatch(submitPaymentMethod(paymentMethod));
```

Comment:
Actually I think you may need to await this as downstream we do `await` the result of this function:

```js
// in useEditableForm.js
 const handleSubmitPaymentsForm = useCallback(
        async formValues => {
            await submitPaymentMethodAndBillingAddress({
                formValues
            });
            setEditing(null);
        },
        [setEditing, submitPaymentMethodAndBillingAddress]
    );
```

Reply:
The async actions return thunks. `await`ing is not doing what we think it is.

```

export const submitPaymentMethodAndBillingAddress = payload =>
    async function thunk(dispatch) { ... }
```

Reply:
The actual function, though named the same, is actually the bound action creator through use of `bindActionCreators` in each context module. It's async :) 

https://redux.js.org/api/bindactioncreators

Reply:
Oh awesome, thanks.

Pausing pending architecture discussion.

Reply:
Resolved by https://github.com/magento/pwa-studio/pull/1993/commits/d36b0d4f0385d26b43237da0381e8fcd5fd10cf1.